### PR TITLE
Check for the existence of input tables for mp=30 FSBM

### PR DESCRIPTION
**Pull request by smileMchen**
KEYWORDS: check_a_mundo, FAST_KHAIN_LYNN_SHPUND

SOURCE: internal

DESCRIPTION OF CHANGES: 
check the existence of input tables for running the fast bin microphysics scheme (mp=30)
(FAST_KHAIN_LYNN_SHPUND)

LIST OF MODIFIED FILES:
M        share/module_check_a_mundo.F

TESTS CONDUCTED: 
1. regression test OK

2. With both SBM files linked in:
```
 FAST SBM: INITIALIZING WRF_HUJISBM
 FAST SBM: ****** WRF_HUJISBM *******
 FAST_SBM_INIT : succesfull reading Table-1                                      
  FAST_SBM_INIT : succesfull reading Table-1
 FAST_SBM_INIT : succesfull reading Table-2                                      
  FAST_SBM_INIT : succesfull reading Table-2
 FAST_SBM_INIT : succesfull reading Table-3                                      
  FAST_SBM_INIT : succesfull reading Table-3
  FAST_SBM_INIT : succesfull reading Table-4
  FAST_SBM_INIT : succesfull reading Table-5
  FAST_SBM_INIT : succesfull reading Table-6
  FAST_SBM_INIT : succesfull reading Table-7
  FAST_SBM_INIT : succesfull reading Table-8
  FAST_SBM_INIT : succesfull reading Table-9
 scattering input directory is source/scattering_tables_2layer_high_quad_1dT_1%fw_110/
 READING SCATTERING TABLES: RAIN
 READING SCATTERING TABLES: SNOW
 READING SCATTERING TABLES: GRAUPEL
 READING SCATTERING TABLES: HAIL
module_mp_WRFsbm : succesfull reading Table-10
  FAST_SBM_INIT : succesfull reading "courant_bott_KS"
 CONCCCNIN   401.74097285574766     
 CONCCCNIN   1799.7710944902556     
  module_mp_WRFsbm : succesfull reading "LogNormal_modes_Aerosol"
  FAST_SBM_INIT : succesfull reading BREAKINIT_KS"
 IKR_Spon_Break=          33
  FAST_SBM_INIT : succesfull reading "Spontanous_Init"
```

3. When the link exists, but the actual file is removed, the logic still works:
```
--- ERROR: Input table SBM_input_33 doesn't exist !!!
--- ERROR: Download this table from http://www2.mmm.ucar.edu/wrf/src/wrf_files/
--- ERROR: Input table scattering_tables_2layer_high_quad_1dT_1%fw_110 doesn't exist !!!
--- ERROR: Download this table from http://www2.mmm.ucar.edu/wrf/src/wrf_files/
-------------- FATAL CALLED ---------------
FATAL CALLED FROM FILE:  <stdin>  LINE:    2194
NOTE:       2 namelist settings are wrong. Please check and reset these options
-------------------------------------------
```

### CODE REVIEW

File: share/module_check_a_mundo.F [hunk 1]
```diff
@@ -569,6 +570,27 @@ END FUNCTION bep_bem_nbui_max
       END DO
 
 #if (EM_CORE == 1)
+!--------------------------------------------------------------------------------------------------
+! Input tables must exist in running directory for fast bin microphysics scheme (mp_physics = 30)
+!--------------------------------------------------------------------------------------------------
+      IF (  model_config_rec % mp_physics(1) .EQ. FAST_KHAIN_LYNN_SHPUND   ) THEN
+           INQUIRE(FILE='./SBM_input_33', EXIST=fsbm_table1_exists)
+           IF (.not.fsbm_table1_exists ) THEN
```
Comment (davegill) [inline share/module_check_a_mundo.F]: @smileMchen 
Ming,
Follow the indenting convention that seems to be in use (looks like 3 spaces).

File: share/module_check_a_mundo.F [hunk 2]
```diff
@@ -569,6 +570,27 @@ END FUNCTION bep_bem_nbui_max
...
+           wrf_err_message = "Input table SBM_input_33 doesn't exist !!!"
+           CALL wrf_debug ( 0, wrf_err_message )
+           wrf_err_message = 'Download this table from http://www2.mmm.ucar.edu/wrf/src/wrf_files/'
+           CALL wrf_debug ( 0, wrf_err_message )
+           count_fatal_error = count_fatal_error + 1
```
Comment (davegill) [inline share/module_check_a_mundo.F]: @smileMchen 
Ming,
Indent these lines and the similar lines in the next IF test.

File: share/module_check_a_mundo.F (hunk 1 above)
Comment (smileMchen) [reply share/module_check_a_mundo.F]: @davegill 
Dave, the indent has been changed to keep consistent with codes below.

File: share/module_check_a_mundo.F (hunk 2 above)
Comment (smileMchen) [reply share/module_check_a_mundo.F]: @davegill 
Dave,
I have changed the indent as suggested. Thanks.

**Comment by weiwangncar** [comment]
@smileMchen @davegill Does this work for linked files?

**Comment by davegill** [comment]
@weiwangncar 
> Does this work for linked files?

Wei,
Yes

**Comment by weiwangncar** [comment]
@davegill What if the link of the file is there, but the file isn't?

**Comment by davegill** [comment]
@weiwangncar 
> What if the link of the file is there, but the file isn't?

Wei,
Good idea. I tested that, too. The Fortran INQUIRE function is smart enough to figure out a busted link == "directory does not exist".

**Pull request merged**